fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4

```diff
@@ -524,6 +526,17 @@ identifierList
     : '(' identifier (',' identifier)* ')'
     ;
 
+identifierOrString
+    : identifier        # identifierOfIdentifierOrString
+    | string            # stringOfIdentifierOrString
+    ;
+
+userIdentifier
+    : identifierOrString                                     # userWithoutHost
+    | identifierOrString '@' identifierOrString              # userWithHost
+    | identifierOrString '@' '[' identifierOrString ']'      # userWithHostAndBlanket
```

Comment:
mysql support this syntax?

Reply:
We have already supported this syntax which refers to user login from this domain. See https://docs.starrocks.com/zh-cn/2.1/sql-reference/sql-statements/account-management/CREATE%20USER#example.